Result/ResultType.swift

```diff
@@ -28,4 +28,31 @@ public extension ResultType {
 	var error: Error? {
 		return analysis(ifSuccess: { _ in nil }, ifFailure: { $0 })
 	}
-}
\ No newline at end of file
+
+	/// Returns a new Result by mapping `Success`es’ values using `transform`, or re-wrapping `Failure`s’ errors.
+	public func map<U>(@noescape transform: Value -> U) -> Result<U, Error> {
+		return flatMap { .Success(transform($0)) }
+	}
```

Comment:
I have philosophical difficulties with a method on `ResultType` producing a `Result`. Unfortunately, we can’t make it return `Self`-parameterized-by-`U` (because Swift lacks higher-ranked types). That’s why I made this a method on `Result` in the first place; presumably if you’re `map`ing some other `ResultType`, then you expect to get an instance of that type back.

(I have the same objection to `SequenceType.map` returning `Array`, but as yet no one has put me in charge of the Swift stdlib.)